text/0003-blockspace-regions.md

```diff
@@ -0,0 +1,197 @@
+  5. **The solution SHOULD minimize the required advance notice for scheduling Coretime.** 
+    - Application load is highly volatile, and the core scheduling primitives should enable applications to pay for only the Coretime they need, while allowing them to gracefully handle periods of heavy load. It is quite common for applications to suddenly experience a large influx of traffic and interest which lasts for a short time.
+  6. **The solution MUST not allow chains to access more than their scheduled amount of Coretime.**
+  7. **The solution MUST NOT allow parachains to build up arbitrary amounts of Coretime to spend later on**
+    - The intention of regions is to ensure consistent rates of utilization by scheduled parachains. Allowing arbitrary amounts of Coretime to be built up and spent later will lead to misallocation of system resources during periods of high demand. Eliminating this type of arbitrage is necessary.
+
+## Stakeholders
+
+- Protocol Reaearchers and Developers.
+- Polkadot Parachain teams both present and future, and their users.
+- DOT Token Holders
+
+These concepts have been alluded to in [Polkadot: Blockspace over Blockchains](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/) and discussed, with previous drafts of the design, [on the Polkadot Forum](https://forum.polkadot.network/t/unifying-bulk-and-mid-term-blockspace-with-strided-regions/2228)
+
+## Explanation
+
+### Parameters:
+
+| Name                | Constant | Value     |
+| ------------------- | -------- | --------- |
+| RATE_DENOMINATOR    | YES      | 5040      |
+| SCHEDULING_LENIENCE | NO       | 16        |
+| HYPERCORES          | NO       | 8         |
+| HYPERCORES_PER_CORE | NO       | 1         |
+
+### Region and RegionSchema
+
+A region is a data structure outlined below:
+
+```rust
+struct RegionSchema {
+    // Relay-chain block number at which this region becomes active.
+    start: u32,
+    // Duration of the region in relay-chain blocks, i.e. it ends at start + duration
+    duration: u32,
+    // The maximum number of blocks which can be made in the region.
+    // `None` signifies that the maximum should be limited only by the duration and rate.
+    maximum: Option<u32>,
+    // This value determines the rate at which the parachain produces blocks.
+    // 
+    // It is the numerator of a fraction where the denominator is `RATE_DENOMINATOR`, where
+    // the fraction indicates the rate at which this region provides blockspace as compared to the
+    // relay-chain's block-rate. That is, a numerator of `RATE_DENOMINATOR`, i.e. 1/1, means that
+    // the region allows blocks to be made every relay chain block. A numerator of `RATE_DENOMINATOR/2` gives
+    // one block every 2 relay-chain blocks. And so on.
+    //
+    // This value may not be greater than `RATE_DENOMINATOR`.
+    rate_numerator: u32,
```

Comment:
Describing like this would appear to prevent the possibility of a ParaId being allocated a fraction above 0.5 of a core's blocks.

Reply:
Fractions such as `1/1`, `3/4`, `5/6`, `2/7`, `7/10` are all possible to normalize to the `RATE_DENOMINATOR`.